menuinst/platforms/win_utils/registry.py

```diff
@@ -72,7 +72,7 @@ def register_file_extension(extension, identifier, command, icon=None, mode="use
         log.debug("Created registry entry for command '%s'", command)
 
         if icon:
-            subkey = winreg.OpenKey(key, identifier)
+            subkey = winreg.OpenKey(key, identifier, access=winreg.KEY_SET_VALUE)
```

Comment:
Awesome catch!

We don't need this anywhere else for other keys, right? Thinking of file type association and so on.

Reply:
The only other key we open is here: https://github.com/conda/menuinst/blob/75f7fa0334d3f363b700904590bcc987c1990cf4/menuinst/platforms/win_utils/registry.py#L47-L52

This is done with write-only access as the default. However, we only write to subkeys and the documentation there is a little confusing:

> The key identified by the key parameter must have been opened with [KEY_SET_VALUE](https://docs.python.org/3/library/winreg.html#winreg.KEY_SET_VALUE) access.

However, when a subkey is not null, the source code suggests that they are always opened with `KEY_SET_VALUE` access: https://github.com/python/cpython/blob/6d419db10c84cacbb3862e2adc940342175bfce9/PC/winreg.c#L1787-L1796

My interpretation is that the reason the icon fails is that we don't explicitly pass a subkey value, but instead operate on the key object itself.

Reply:
Got it. So for now no action needed but let's keep an eye for reports in case we need to pass more access flags. Thanks for the research!